### PR TITLE
added the controllers

### PR DESCRIPTION
**Pull request by EddyShimwa**
### Hello, In this PR I,
- Created the controllers for the users and posts
- Created the views and actions for my controllers

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -1,7 +1,7 @@
 source 'https://rubygems.org'
 git_source(:github) { |repo| "https://github.com/#{repo}.git" }
 
-ruby '3.0.5'
+ruby '3.2.2'
 
 # Bundle edge Rails instead: gem "rails", github: "rails/rails", branch: "main"
```
Comment (ShadyShawkat) [inline Gemfile:6]: - [x] [OPTIONAL] Kindly make sure the seeds are working properly. Currently, when I tried to set up the database with the seed, I got the following error.

image

 - If you want to simulate getting this error, try dropping the databases and setting them up again and the error should happen.

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class UsersController < ApplicationController
+  def show
+    @user = User.find(params[:id])
+  end
+end
```
Comment (ShadyShawkat) [inline app/controllers/users_controller.rb]: - [x] Please make sure that the `UsersController` has an `index` action as well, as per the given routes in the requirements.
https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/routes.png

File: app/controllers/posts_controller.rb [hunk 3]
```diff
@@ -0,0 +1,6 @@
+class PostsController < ApplicationController
+  def show
+    @user = User.find(params[:user_id])
+    @post = Post.find(params[:id])
+  end
+end
```
Comment (ShadyShawkat) [inline app/controllers/posts_controller.rb]: - [x] Please make sure that the `PostsController` has an `index` action as well, as per the given routes in the requirements.
https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/routes.png

**Review by ShadyShawkat — CHANGES_REQUESTED** [review]
## Hi @EddyShimwa, 

Good job so far! 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight
- All linters are passing ✔️ 
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 

Image

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 3 above)
Comment (EddyShimwa) [reply app/controllers/posts_controller.rb]: thanks @ShadyShawkat

File: app/controllers/users_controller.rb (hunk 2 above)
Comment (EddyShimwa) [reply app/controllers/users_controller.rb]: Thanks @ShadyShawkat

**Review by VICTOR-CHUKWU — APPROVED** [review]
Hi @EddyShimwa , 
# STATUS: APPROVED! 🥇 💯 
done

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by EddyShimwa** [comment]
Thanks @VICTOR-CHUKWU  fro approving my PR

**Pull request merged**